login_init.zsh

```diff
@@ -23,32 +23,23 @@
     zrecompile -pq ${dir}.zwc ${dir}/^([_.]*|prompt_*_setup|README*|*.zwc|*.zwc.old)(-.N)
   done
 
-  # zcompile enabled module init scripts
+  # zcompile enabled module init scripts and subdirs
+  setopt nullglob
   for zmodule (${zmodules}); do
     zmodule_dir=${ZIM_HOME}/modules/${zmodule}
     if [[ -d ${zmodule_dir} ]]; then
-      for file (${zmodule_dir}/init.zsh \
-          ${zmodule_dir}/{,zsh-}${zmodule}.{zsh,plugin.zsh,zsh-theme,sh}); do
+      for file (${zmodule_dir}/**/*.{zsh,sh,zsh-theme} \
+          ${zmodule_dir}/*.{zsh,sh,zsh-theme}); do
```

Comment:
If we want to be less specific here, I think the glob should be `${zmodule_dir}/(^*test*/)#*.{zsh,zsh-theme,sh}(.N)` -- don't match `*test*` directories (see https://github.com/zsh-users/zsh-syntax-highlighting/pull/443 and #205), only match plain files, and enable NULL_GLOB in this glob (no need to do setopt and unsetopt outside).

@Eriner, what do you think? This will potentially make login_init take more time zcompiling more files.

Reply:
I don't have an issue with the zcompiling taking longer - even if it takes 2m (spoiler alert: it won't) on first run, who cares? It's smart enough that it won't redundantly zcompile unecessarily, so that penalty will only happen once and as a result every subsequent shell will load faster. Sounds like a good trade-off to me.